pyqtgraph/graphicsItems/PlotDataItem.py

```diff
@@ -38,6 +38,85 @@ class PlotDataItem(GraphicsObject):
     sigClicked = QtCore.Signal(object, object)
     sigPointsClicked = QtCore.Signal(object, object, object)
     sigPointsHovered = QtCore.Signal(object, object, object)
+    
+    class PlotDataset(object):
+        """ Holds collected information for a plotable dataset """
+        def __init__(self, x, y):
+            super().__init__()
+            assert x is not None
+            assert y is not None
```

Comment:
`asset` statements in functional code is generally a bad idea; they are completely ignored in the case of `python -O` so best to keep them strictly for test code.  If we want checks here; probably should raise a `RuntimeError`.

Reply:
These were added to test if any part of the test suite manages to instantiate a PlotDataset with data set to None.
To avoid replicating unneeded code everywhere, the convention throughout PlotDataItem is that a PlotDataset will only be instantiated when there is actual data. Otherwise the reference is kept as None instead.

I think this can simply be removed. I added comments to the code explaining the convention, which should only be relevant to developers working inside PlotDataItem.

Reply:
I'm good w/ removing them 👍🏻